### PR TITLE
Add FedProx algorithm implementation

### PR DESCRIPTION
**Pull request by itrushkin**
- [x] PyTorch implementation
- [x] Keras implementation
- [x] Workspaces

FedProx paper: https://arxiv.org/pdf/1812.06127.pdf

### CODE REVIEW

File: openfl/federated/task/metric.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+"""Metric module."""
+
+from typing import NamedTuple
+import numpy as np
+
+
+class Metric(NamedTuple):
```
Comment (psfoley) [inline openfl/federated/task/metric.py:7]: The definition for `Metric` should probably be kept in `openfl.utils.types` where `TensorKey` and `TaskResultKey` is defined

**Review by psfoley — APPROVED** [review]

**Comment by alexey-gruzdev** [comment]
Jenkins please retry a build
